src/web/handlers.rs

```diff
@@ -416,6 +394,54 @@ pub async fn post_collection_batch(
         return Err(ApiError::from(err).into());
     };
 
+    // write the BSOs contained in the commit request.
+    if !coll.bsos.valid.is_empty() {
```

Comment:
should only happen on commit:

```suggestion
    if commit && !coll.bsos.valid.is_empty() {
```

Reply:
I agree, that's why I was asking about the change to line 321 which handles the "not commit" messages. That should capture all non-commit messages, no? Meaning that anything that got as far as here should be a commit. I could make things a bit clearer by wrapping this whole section in an `else` for `!breq.commit`, but I'm not sure that matches the style we've used before.